pyCGM_Single/tests/test_Pipelines.py

```diff
@@ -0,0 +1,423 @@
+        staticData = loadData(static_trial)
+        if (calcSacrum):
+            for frame in motionData:
+                frame['SACR'] = pelvisJointCenter(frame)[2]
+        data = dataAsDict(motionData, npArray=True)
+        static = dataAsDict(staticData, npArray=True)
+        return data, static
+
+    def test_butterFilter(self):
+        tests = [[-1003.58361816, -1003.50396729, -1003.42358398, -1003.3425293,
+                  -1003.26068115, -1003.17810059, -1003.09484863, -1003.01080322,
+                  -1002.92608643, -1002.84063721, -1002.75445557, -1002.6675415,
+                  -1002.57995605, -1002.49157715, -1002.40252686, -1002.31274414], #normal case
+                 [70, 76, 87, 52, 28, 36, 65, 95, 69, 62, 60, 17, 12, 97, 11, 48], #test ints
+                 [-70, 76, 87, -52, 28, 36, 65, 95, -69, 62, 60, 17, 12, -97, 11, 48]] #test ints, mixed positive and negative
+        cutoff = 20
+        Fs = 120
+        expectedResults = [np.array([-1003.58359202, -1003.50390817, -1003.42363665, -1003.34252947,
+                                   -1003.26066539, -1003.17811173, -1003.09483484, -1003.01081474,
+                                   -1002.92608178, -1002.8406421 , -1002.7544567 , -1002.66753023,
+                                   -1002.57993488, -1002.49164743, -1002.40251014, -1002.31267097]),
+                          np.array([69.98542399, 84.63525626, 78.90545928, 52.35480234, 
+                                    30.49005341, 37.66166803, 65.18623797, 84.35193788, 
+                                    82.07291528, 64.17505539, 41.24155139, 28.10878834,
+                                    33.98739195, 47.41539095, 52.2509855,  48.00079742]),
+                          np.array([-69.96417558, 62.8459697, 68.01436426, -1.36719047,
+                                    -12.10645143, 48.64339139,  81.14625417,  41.74781614,
+                                    1.18057508, 21.16904888, 57.9718148, 39.85271513, 
+                                   -21.05573085, -53.81198321, -23.50929002, 47.94081346])]
+        for i in range(len(tests)):
+            data = tests[i]
+            numpyData = np.array(tests[i]) #Tests to make sure that the input can also be a numpy array
+            result = Pipelines.butterFilter(data, cutoff, Fs)
+            numpyResult = Pipelines.butterFilter(numpyData, cutoff, Fs)
+            expectedResult = expectedResults[i]
+            for i in range(len(result)):
+                np.testing.assert_almost_equal(result[i], expectedResult[i], self.rounding_precision)
+                np.testing.assert_almost_equal(numpyResult[i], expectedResult[i], self.rounding_precision)
+
+        #Test that if len(data) < 15, an exception is raised
+        dataLenTests = [[],
+                        [0, 1, 2, 3], 
+                        [0.1, 0.2, 0.3], 
+                        [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14]]
+        for test in dataLenTests:
```

Comment:
Please test one thing at a time

Reply:
Could you clarify which things I am testing at the same time? I wrote this to test four different instances where the length of data is too short and made sure that an exception was raised.